types/schema/data/Property.ts

```diff
@@ -96,20 +98,40 @@ type PropertyTypeMap = {
  */
 export type PropertyType = PropertyTypeMap[keyof PropertyTypeMap];
 
-/** @todo in future value & description should check against PlanningConstraints enum, but also allow custom per-council variables ?? */
-interface BasePlanningConstraint {
-  value: string;
-  description: string;
-}
+type PlanningConstraintKeys = keyof typeof PlanningConstraints;
 
-interface NonOverlappingPlanningConstraint extends BasePlanningConstraint {
+type GenericPlanningConstraint<TKey extends PlanningConstraintKeys> = {
+  value: TKey;
+  description: (typeof PlanningConstraints)[TKey];
+  category?:
+    | 'Ecology'
+    | 'Flooding'
+    | 'General policy'
+    | 'Heritage and conservation'
+    | 'Trees';
+};
+
+type PlanningConstraintMap = {
+  [K in PlanningConstraintKeys]: GenericPlanningConstraint<K>;
+};
+
+type BasePlanningConstraint =
+  PlanningConstraintMap[keyof PlanningConstraintMap];
+
+type NonOverlappingPlanningConstraint = {
   overlaps: false;
-}
+} & BasePlanningConstraint;
 
-interface OverlappingPlanningConstraint extends BasePlanningConstraint {
+type OverlappingPlanningConstraint = {
   overlaps: true;
-  entities: string[];
-}
+  entities:
+    | {
+        name: string;
+        description?: string;
+        source?: URL;
```

Comment:
Based on outcome of [this conversation](https://opendigitalplanning.slack.com/archives/C05NQJK5S9L/p1696928598458159), I imagine `entities` will grow to have an optional `rules` property or similar in the future !